amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/DefaultTableService.java

```diff
@@ -251,32 +255,94 @@ public Blocker block(
       TableIdentifier tableIdentifier,
       List<BlockableOperation> operations,
       Map<String, String> properties) {
-    checkStarted();
-    return getAndCheckExist(getOrSyncServerTableIdentifier(tableIdentifier))
-        .block(operations, properties, blockerTimeout)
-        .buildBlocker();
+    Preconditions.checkNotNull(operations, "operations should not be null");
+    Preconditions.checkArgument(!operations.isEmpty(), "operations should not be empty");
+    Preconditions.checkArgument(blockerTimeout > 0, "blocker timeout must > 0");
+    String catalog = tableIdentifier.getCatalog();
+    String database = tableIdentifier.getDatabase();
+    String table = tableIdentifier.getTableName();
+    int tryCount = 0;
+    while (tryCount++ < 3) {
+      long now = System.currentTimeMillis();
+      doAs(
+          TableBlockerMapper.class,
+          mapper -> mapper.deleteExpiredBlockers(catalog, database, table, now));
```

Comment:
I am not sure if it is necessary to delete expired blockers every time blocking is called.
It may increase the cost time.

Reply:
I think it doesn't matter here. 
1. TableBlocker is a low-frequency operation. 
2. There is a TableBlockerExpire thread. Actually, not too much data will be deleted in this place. 
3. There are indexes for catalog, db, and table, and the query efficiency won't be too low.